### ISSUE TITLE
Sending Proactive Message to Facebook user with Message Tag

### ISSUE THREAD

**Issue by bhushanvadgave** [microsoft/botbuilder-js#3521]
## Versions
NodeJS 10.16
BotBuilder NodeJS SDK 4.8.0
Channel: Facebook Messenger 
Facebook App ID: 1366431413469216 

## Context
Our bot needs to send proactive messages (notification) to Facebook users, **outside of 24-hour window**. We are using `channelData` property of Activity to include `messaging_type` and `tag` data while sending the message.

## Description of the bug
Below is the the sample activity object before `await turnContext.sendActivity(activity)` is called.
```
{
  type: "message",
  text: "Hello, this is a test notification",
  inputHint: "acceptingInput",
  channelData: {
    isNotificationMsg: true,
    facebook: {
      messaging_type: "MESSAGE_TAG",
      tag: "ACCOUNT_UPDATE"
    },
    messaging_type: "MESSAGE_TAG",
    tag: "ACCOUNT_UPDATE",
    convSessionId: "gsTVPjZVatUXSNBK"
  }
}
```
_Note: We were not sure whether to put `messaging_type` and `tag` inside `facebook` field or not, so we have kept them at both the places._

But message is not getting delivered to the user. The proactive message sending works if tried withing 24-hour window.
There are no errors appearing in out bot service's server logs or "Channels" screen of Bot Registration in Azure. 
Also we couldn't find a way to see the final message that Microsoft Bot Connector forms before sending it to Facebook APIs.

**Comment by joshgummersall** [comment]
Take a look at the [`channelData` documentation](https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-connector-channeldata?view=azure-bot-service-4.0#create-a-facebook-notification). In particular, it looks like `channelData` is provided directly to the service in question. According to [the Facebook documentation](https://developers.facebook.com/docs/messenger-platform/send-messages/message-tags#sending), the `channelData` likely should not include unknown keys. Try removing everything but the `messaging_type` and `tag` keys to see if that resolves the issue.

**Comment by bhushanvadgave** [comment]
Sorry for the delayed reply.
@joshgummersall Tried sending the message this time with only the required channelData fields. But still, it did not work.
```
{
  type: "message",
  text: "Hello, this is a test notification",
  inputHint: "acceptingInput",
  channelData: {
    "notification_type": "REGULAR",
    "messaging_type": "MESSAGE_TAG",
    "tag": "ACCOUNT_UPDATE"
  }
}
```

**Comment by mdrichardson** [comment]
@bhushanvadgave I'm still digging into this,  but can you confirm that you don't have any Policy Violation Notifications? Per [this page (at the bottom)](https://developers.facebook.com/docs/messenger-platform/policy/policy-overview#24hours_window), you'd go to Page Settings > Page Support Inbox.

**Comment by bhushanvadgave** [comment]
@mdrichardson 
There are no notifications in Page Support Inbox.
Also tried sending the message with the below curl command and **it worked. The user received the message.**
```
curl -X POST -H "Content-Type: application/json" -d '{
  "recipient": { 
    "id": "<PSID>"
  },
  "message":{
    "text": "Hello, this is a test notification"
  },
  "messaging_type": "MESSAGE_TAG",
  "tag": "ACCOUNT_UPDATE"
      
}' "https://graph.facebook.com/v10.0/me/messages?access_token=<PAGE_ACCESS_TOKEN>"
```
If tried above command without `messaging_type` as `MESSAGE_TAG` it gives `(#10) This message is sent outside of allowed window.` error. So it looks like there is no issue with the message text or Facebook API. It seems there is some issue at Microsoft connector.

Just FYI user's last message time is more than a week old.

**Comment by mdrichardson** [comment]
@bhushanvadgave I agree; this is sounding like an issue with the Facebook Channel. Can you please:

1. Confirm that you're only using the Facebook Channel and not the [Facebook Adapter](https://botkit.ai/docs/v4/platforms/facebook.html)
2. Provide your bot's AppId. We can look into our backend logs and see if anything stands out. Feel free to email it to me, if you prefer; my email address is in my profile.

**Comment by bhushanvadgave** [comment]
@mdrichardson 
Yes, we are using the Facebook channel and not using Facebook Adapter.
Bot App Id: b08d39ff-0815-4f73-9ee6-56c256454a23

**Comment by tdurnford** [comment]
@arturl Do we support message tags in the Facebook Channel?
